File: python/examples/getting-started/00_your_first_call.py
# Basic function execution with call()
from opperai import ModelConfig, Opper

opper = Opper()

result = opper.call("my-first-call", input="In one sentence, What is Python?")

print("Full result object:", result)

# You can also specify a model — as a string or with provider-specific options
result2 = opper.call(
    "my-first-call",
    input="In one sentence, What is Python?",
    model=ModelConfig(name="anthropic/claude-sonnet-4-6", options={"max_tokens": 100}),
)

print("With model options:", result2.data)

File: typescript/examples/getting-started/00-your-first-call.ts
// Basic function execution with call()
import { Opper } from "../../src/index.js";

const opper = new Opper();

const result = await opper.call("my-first-call", {
  input: "What is TypeScript?",
});

console.log("Full result object:", result);

// You can also specify a model — as a string or with provider-specific options
const result2 = await opper.call("my-first-call", {
  input: "What is TypeScript?",
  model: { name: "anthropic/claude-sonnet-4-6", options: { max_tokens: 100 } },
});

console.log("With model options:", result2.data);
